openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -382,16 +569,35 @@ def extract_geo_data(media_data):
                 geo_data[key] = key_value
         return geo_data
 
+    def extract_global_usage(self, media_data) -> int:
+        """
+        Extract the global usage count from the media data.
+        The global usage count serves as a proxy for popularity data, since it
+        represents how many pages across all available wikis a media item is used on.
+        This uses a cache to record previous values for a given foreign ID, because
+        it is possible that we might encounter a media item several times based on the
+        querying method used (see "props" above). Not all results will have the same
+        (if any) usage data, so we want to record and cache the highest value. Most
+        items have no global usage data, so we only cache items that have a value.
```

Comment:
> it is possible that we might encounter a media item several times based on...

Maybe I'm missing something else but does this mean that we get duplicates from the API? 🤔 How are they handled? Also, given this cache is not using Redis or something similar, although it's just for results with global usage data, it seems still a potentially high numbers of registries to save in memory, per the size of Wikimedia images, our second biggest provider (or perhaps now it's the third one after iNatualist?) Anyway, it's worth giving it a try!

Reply:
It does not, we filter out duplicates as part of the TSV loading step: 

https://github.com/WordPress/openverse-catalog/blob/3ab5e572b061180250bfee9741d95c813ce8f7f7/openverse_catalog/dags/common/loader/sql.py#L219-L231

This cache is only per-ingestion run, and it looks like our average run is ingesting about ~15k items. So even if every item had popularity data, the cache would only have 15k records, which I think is okay! Definitely within normal running parameters for Python IMO 😄

Reply:
Good to know about the limited number of rows cached! However, for what I can tell from that statement, it's not guaranteed that it will keep the row with the highest "global data usage" value. The records are assumed to be identical and take the first one found. So it's not something critical but I wanted to make sure we're all aware of the situation.

Reply:
Ahh you're totally right, it currently just grabs the highest value from whatever's available:

https://github.com/WordPress/openverse-catalog/blob/c476f923c1ace2ab5f0ef40ff552f9516dba49d3/openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py#L586

So there could be a case potentially where one run gathers a _different but smaller_ set of pages which would actually make the total count higher than the maximum value between that and the cache. But already we're throwing out some popularity data for these records since we don't finish processing them 😅 good thing to be aware of for sure!